### PR TITLE
Remove some MIPS-specific changes to top-level build glue

### PR DESCRIPTION
**Pull request by bsdjhb**

### CODE REVIEW

**Review by brooksdavis — APPROVED** [review]
Looks good. The removal of the CHERI variable reminded me to archive the bsdtools repo.

File: Makefile [hunk 1]
```diff
@@ -268,25 +268,7 @@ SUB_MAKE= `test -x ${MYMAKE} && echo ${MYMAKE} || echo ${MAKE}` \
 SUB_MAKE= ${MAKE} -m ${.CURDIR}/share/mk
 .endif
 
-.if defined(CHERI)
-TARGET=		mips
-TARGET_ARCH=	mips64
-CHERI_FLAGS=	-DDB_FROM_SRC \
-		-DMALLOC_PRODUCTION \
-		LOCAL_DIRS="ctsrd tools/tools/atsectl" \
```
Comment (jrtc27) [inline Makefile:276]: We should retire ctsrd too, the only stuff there is DE4 bits that I don't think have been used for many years even whilst DE4s were in use (though could be wrong about that last part)

**Review by jrtc27 — APPROVED** [review]
Glad to see this go, nearly axed it myself when cleaning up bsd.cheri.mk but didn't, I think because of concerns that dropping -DMALLOC_PRODUCTION might be a bad idea if people running DE4 benchmarks weren't aware it had gone away (but it did remind me that we needed to set the magic symlink in /etc to mostly recover that performance for our CHERI-RISC-V benchmarking jobs), plus not being entirely sure about the state of ctsrd.

File: Makefile.inc1 [hunk 2]
```diff
@@ -1083,17 +1083,6 @@ _cleanobj_fast_depend_hack: .PHONY
 	@echo ">>> Deleting stale dependencies...";
 	sh ${.CURDIR}/tools/build/depend-cleanup.sh ${OBJTOP}
 
-.if ${TARGET:Mmips*}
```
Comment (jrtc27) [inline Makefile.inc1:1086]: Somehow I think we could have retired this a while ago...

Comment (bsdjhb) [reply Makefile.inc1:1086]: Yes, most likely.

File: Makefile (hunk 1 above)
Comment (bsdjhb) [reply Makefile:276]: I pushed commits to remove these two dirs as these are the only references I could find to them.

**Pull request merged**